decentraland/unity-explorer
# fix: character trigger area + avatar modifier area

## What does this PR change?

Fixes https://github.com/decentraland/unity-explorer/issues/2815

The issue here was that we are now spawning scenes at -10.000 in y coord, so when the assets are loaded, their cached positions are also set with that y and unless the transform changes we dont update it. AND the trigger areas copy their position each cycle from the CachedTransform in the TransformComponent.
The fix for that will go in another PR (which requires recursively update all the cached transforms of all the children of the scene after we set it in its definite position) as I found as simpler solution for this particular issue.
Parenting! By parenting the Trigger Area to the entity that in theory contains it, we ensure proper movement without needing any constant updating of the transform position to mimic any potential change in the parent's transform. After all, that's what parenting is for.

## How to test the changes?

#### TEST A
Log into metadynelabs. The camera should be forced into 1st person and you should not see any other players around you.
That's all :)

#### TEST B
1. Download this scene: https://github.com/decentraland/sdk7-test-scenes/tree/main/scenes/9%2C99-modifier-areas and uncompress it somewhere
2. Enter the scene root folder and run `npm i` and then `npm run start -- --explorer-alpha`
4. Close the Explorer that auto-opened. Leave the scene running in the console/terminal.
5. Download the build from this PR and connect it to the running scene using [the console command](https://github.com/decentraland/unity-explorer/wiki/How-to-connect-to-a-local-scene-(Unity-Editor---Custom-Build---Latest-Released-Build)#connecting-a-custom-build-to-the-scene) according to your OS.
6. Enter the 2 "Camera mode area" areas and confirm any of them changes your camera mode (the left one changes to first person cam, and the right one changes to 3rd person cam)
